### PR TITLE
Revision by date

### PR DESCRIPTION
**Pull request by noamross**
Implements downloading a revision by date, as well as automatic metadata lookup in `download_revision()`.

Quick edit to `download_file()`, to complete overwriting by default.

Part of UI improvements for #4.

No tests yet.  Can we create some open/shared google docs for a test suite?

### CODE REVIEW

File: R/revisions.R [hunk 1]
```diff
@@ -94,21 +94,56 @@ delete_revision <- function(token, file_id, rev_id, ...){
...
 #'@param ... any further arguments to pass to httr's GET.
 #'
 #'@return TRUE if the file could be downloaded, FALSE or an error otherwise.
 #'@export
-download_revision <- function(token, metadata, download_type, destination, ...){
+download_revision <- function(token, metadata=NULL, download_type, destination, file_id=NULL, version = NULL, overwrite=TRUE, ...) {
+  
+  if(is.null(metadata) & is.null(file_id)) stop("file_id or metadata must be provided.")
+  
+  if(is.null(file_id)) file_id = detect_full_url(file_id)
```
Comment (Ironholds) [inline R/revisions.R]: Opening parens? :)

File: R/revisions.R [hunk 2]
```diff
@@ -94,21 +94,56 @@ delete_revision <- function(token, file_id, rev_id, ...){
...
+#'by default.
+#'
 #'@param ... any further arguments to pass to httr's GET.
 #'
 #'@return TRUE if the file could be downloaded, FALSE or an error otherwise.
 #'@export
-download_revision <- function(token, metadata, download_type, destination, ...){
+download_revision <- function(token, metadata=NULL, download_type, destination, file_id=NULL, version = NULL, overwrite=TRUE, ...) {
+  
+  if(is.null(metadata) & is.null(file_id)) stop("file_id or metadata must be provided.")
```
Comment (Ironholds) [inline R/revisions.R]: Could you follow the code conventions in the repo as a whole?
if(is.null(metadata) & is.null(file_id)){
    stop(...)
}

File: R/revisions.R [hunk 3]
```diff
@@ -94,21 +94,56 @@ delete_revision <- function(token, file_id, rev_id, ...){
...
+  
+  if(is.null(file_id)) file_id = detect_full_url(file_id)
+  
+  if(class(metadata) != "rev_metadata") {
+    if(class(metadata) == "file_metadata") file_id = metadata$id
+    
+    version_ = try(as.POSIXct(version), silent=TRUE)
+    if(class(version) == "try-error") version_ = version
+    
+    file_id = detect_full_url(file_id)
```
Comment (Ironholds) [inline R/revisions.R]: Conventions again; <- over =

**Comment by Ironholds** [comment]
Thanks! Comments added. Additional-meta-comment; could you add a unit test testing these capabilities?

**Comment by noamross** [comment]
Will do.  Are we assuming that all tests should be done on the first document in the documents list of whatever account the token is for on Travis?

**Comment by Ironholds** [comment]
Pretty much. As someone more informed about travis than I, can you tell me if it'd be secure and viable to use your encrypted token rather than mine? We're running into differing behaviour around refreshing, causing everything ever to break - I'm wondering if it's down to how google has my driver instance registered, orrr..

**Comment by noamross** [comment]
It may take me a couple of days to get back to this, but I'll make a
Travis-CI instance for my fork and see if my tokens help anything.  Right
now it looks like there's something wrong with the token getting decrypted
altogether, so the build is erroring out before tests even run.  But yeah,
the specific permissions set in your google app could be an issue.

File: R/revisions.R (hunk 1 above)
Comment (noamross) [inline R/revisions.R]: OK, I'm on it.  What did you mean here?

Comment (Ironholds) [inline R/revisions.R]: I...actually have no idea. Huh! Ignore it ;p

**Comment by noamross** [comment]
Code conventions adhered to, tests provided.

**Comment by noamross** [comment]
> Speaking of which, assignment looks stupid. I shit you not; these all have the equivalent effect of storing the value of b in a: a <- b; b -> a; assign("a", b); a = b;. There are subtle differences and some authorities prefer <- for assignment but I'm increasingly convinced that they are wrong; = is always safer. R doesn't allow expressions of the form if (a = testSomething()) but does allow if (a <- testSomething()). 
-- Tim Smith, [*AaRrgh: a newcomer's (angry) guide to R*](http://tim-smith.us/arrgh/syntax.html)

**Comment by Ironholds** [comment]
Tim Smith's essay is great ("to a first-order approximation, zero software engineers") but he and I disagree on that bit (although I appreciate the safety point. Didn't know that! Not...entirely sure why anyone would write that, but..)

**Comment by Ironholds** [comment]
Oh, actually, you did miss one important bit, but I'll add that

**Pull request merged**